src/v/cloud_storage/remote.cc

```diff
@@ -715,6 +730,82 @@ ss::future<upload_result> remote::delete_objects(
   std::vector<cloud_storage_clients::object_key> keys,
   retry_chain_node& parent) {
     ss::gate::holder gh{_gate};
+
+    if (!is_batch_delete_supported()) {
+        vlog(
```

Comment:
maybe move this block into a function like `delete_objects_fallback` or similar?  (to keep the existing function reasonably short)

Reply:
fixed